### PR TITLE
Write timing information to an xml file.

### PR DESCRIPTION
**Pull request by markdewing**
It uses the same format as the QMCPACK info.xml file.
The XML handling is done using the TinyXML2 library, which is included in this commit and adds no extra external dependencies.

The file is named with the tiling (e.g. info_1_1_1.xml) to make it easier for the results of differently sized runs to coexist.

The output is enhanced with respect to the QMCPACK info file:
 - The particle size information is output (number of electrons).  The same output should be in the QMCPACK info file soon.
 - The driver name and number of steps and substeps. Similiar information should go into the QMCPACK info file, but the form may differ.

The hardware section is not yet populated (number of OMP threads, etc.)

### CODE REVIEW

**Comment by grahamlopez** [comment]
Is it okay to leave this PR open for a little bit for me to do some CI testing?

**Comment by markdewing** [comment]
Yes, it's fine to leave this PR open for a while.

**Comment by grahamlopez** [comment]
test this please

File: src/Utilities/XMLWriter.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+//    Argonne National Laboratory
+////////////////////////////////////////////////////////////////////////////////
+
+/** @file XMLWriter.h
+ */
+
+#ifndef QMCPLUSPLUS_XML_WRITER_H
+#define QMCPLUSPLUS_XML_WRITER_H
+
+#include <tinyxml/tinyxml2.h>
```
Comment (grahamlopez) [inline src/Utilities/XMLWriter.h]: shall we put the 'tinyxml' directory under 'Utilities'?

Comment (markdewing) [reply src/Utilities/XMLWriter.h]: That would get it out of the top-level src/ directories.

**Pull request merged**